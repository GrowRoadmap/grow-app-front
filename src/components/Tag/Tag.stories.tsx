import React from 'react';
import { Tag } from './Tag';

export default {
    title: 'Tag',
};

export const StatusColors = () => (
    <div>
        <Tag />
        <Tag truncate={{ width: 100 }}>This is an exemple how to truncate</Tag>
        <Tag label='Easy' />
        <Tag label='Medium' />
        <Tag label='Hard' />
        <Tag label='Expert' />
    </div>
);

export const CustomContent = () => (
    <div>
        <Tag truncate={{ width: 100 }}>This is an exemple how to truncate</Tag>
        <Tag label='Disabled'>ðŸ§® This is an exemple how to truncate</Tag>
    </div>
);
