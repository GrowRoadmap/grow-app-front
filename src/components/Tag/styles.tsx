import styled, { css } from 'styled-components';
import { switchProp, theme } from 'styled-tools';
import { Box } from 'components/Box';

export const CircleSign = styled(Box)<any>`
    width: 6px;
    height: 6px;
    border-radius: 100%;
    margin-right: 6px;
    ${switchProp('colorStatus', {
        Disabled: css`
            display: none;
        `,
        Undefined: css`
            background-color: ${theme('colors.levels.undefined')};
        `,
        Easy: css`
            background-color: ${theme('colors.levels.easy')};
        `,
        Medium: css`
            background-color: ${theme('colors.levels.medium')};
        `,
        Hard: css`
            background-color: ${theme('colors.levels.hard')};
        `,
        Expert: css`
            background-color: ${theme('colors.levels.expert')};
        `,
    })}
`;
