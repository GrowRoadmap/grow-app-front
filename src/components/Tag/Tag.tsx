import React from 'react';
import { Box } from 'components/Box';
import { Text } from 'components/Text';
import { CircleSign } from './styles';
import { TruncateProps } from 'react-truncate';

export type TagProps = {
    label?: 'Disabled' | 'Undefined' | 'Easy' | 'Medium' | 'Hard' | 'Expert';
    truncate?: TruncateProps | boolean;
    children?: React.ReactNode;
};

export const Tag: React.FC<TagProps> = ({
    label = 'Undefined',
    truncate,
    children,
}) => (
    <Box
        data-testid='Tag'
        backgroundColor='gray.70'
        borderLeft='2px gray'
        borderRadius='sm'
        p='2'
        display='flex'
        alignItems='center'
        width='fit-content'
        my='2'
    >
        <CircleSign colorStatus={label} />
        <Text color='blueGray' fontSize='sm' m='0' truncate={truncate}>
            {children || label}
        </Text>
    </Box>
);
