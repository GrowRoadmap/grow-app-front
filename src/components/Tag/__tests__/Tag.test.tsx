import React from 'react';
import { render } from '../../../../tests';
import { Tag } from '../Tag';

const text = 'Its a button';

describe('<Tag />', () => {
    it('should render correctly basic variations', () => {
        const { getByText } = render(
            <div>
                <Tag />
                <Tag label='Easy' />
                <Tag label='Medium' />
                <Tag label='Hard' />
                <Tag label='Expert' />
            </div>
        );

        expect(getByText('Undefined')).toBeTruthy();
        expect(getByText('Easy')).toBeTruthy();
        expect(getByText('Medium')).toBeTruthy();
        expect(getByText('Hard')).toBeTruthy();
        expect(getByText('Expert')).toBeTruthy();
    });

    it('should match snapshot', () => {
        const { container } = render(
            <div>
                <Tag />
                <Tag label='Easy' />
                <Tag label='Medium' />
                <Tag label='Hard' />
                <Tag label='Expert' />
                <Tag truncate>This is an exemple how to truncate</Tag>
                <Tag label='Disabled'>
                    ðŸ§® This is an exemple how to truncate
                </Tag>
            </div>
        );
        expect(container).toMatchSnapshot();
    });
});
