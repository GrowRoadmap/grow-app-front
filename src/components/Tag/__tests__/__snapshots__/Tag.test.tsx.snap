// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should match snapshot 1`] = `
<div>
  <div>
    <div
      class="Box-sc-16mv5fl-0 dqYHOz"
      data-testid="Tag"
      display="flex"
      width="fit-content"
    >
      <div
        class="Box-sc-16mv5fl-0 styles__CircleSign-sc-1pvkvyn-0  bozZeK"
      />
      <div
        class="Box-sc-16mv5fl-0 Text__TextStyled-sc-1eacmib-0 cfMSYc gjXGAy"
        color="blueGray"
        font-size="sm"
      >
        Undefined
      </div>
    </div>
    <div
      class="Box-sc-16mv5fl-0 dqYHOz"
      data-testid="Tag"
      display="flex"
      width="fit-content"
    >
      <div
        class="Box-sc-16mv5fl-0 styles__CircleSign-sc-1pvkvyn-0  hLQKiF"
      />
      <div
        class="Box-sc-16mv5fl-0 Text__TextStyled-sc-1eacmib-0 cfMSYc gjXGAy"
        color="blueGray"
        font-size="sm"
      >
        Easy
      </div>
    </div>
    <div
      class="Box-sc-16mv5fl-0 dqYHOz"
      data-testid="Tag"
      display="flex"
      width="fit-content"
    >
      <div
        class="Box-sc-16mv5fl-0 styles__CircleSign-sc-1pvkvyn-0  cCUHJI"
      />
      <div
        class="Box-sc-16mv5fl-0 Text__TextStyled-sc-1eacmib-0 cfMSYc gjXGAy"
        color="blueGray"
        font-size="sm"
      >
        Medium
      </div>
    </div>
    <div
      class="Box-sc-16mv5fl-0 dqYHOz"
      data-testid="Tag"
      display="flex"
      width="fit-content"
    >
      <div
        class="Box-sc-16mv5fl-0 styles__CircleSign-sc-1pvkvyn-0  iWNvsN"
      />
      <div
        class="Box-sc-16mv5fl-0 Text__TextStyled-sc-1eacmib-0 cfMSYc gjXGAy"
        color="blueGray"
        font-size="sm"
      >
        Hard
      </div>
    </div>
    <div
      class="Box-sc-16mv5fl-0 dqYHOz"
      data-testid="Tag"
      display="flex"
      width="fit-content"
    >
      <div
        class="Box-sc-16mv5fl-0 styles__CircleSign-sc-1pvkvyn-0  hTbgPS"
      />
      <div
        class="Box-sc-16mv5fl-0 Text__TextStyled-sc-1eacmib-0 cfMSYc gjXGAy"
        color="blueGray"
        font-size="sm"
      >
        Expert
      </div>
    </div>
    <div
      class="Box-sc-16mv5fl-0 dqYHOz"
      data-testid="Tag"
      display="flex"
      width="fit-content"
    >
      <div
        class="Box-sc-16mv5fl-0 styles__CircleSign-sc-1pvkvyn-0  bozZeK"
      />
      <div
        class="Box-sc-16mv5fl-0 Text__TextStyled-sc-1eacmib-0 cfMSYc gjXGAy"
        color="blueGray"
        font-size="sm"
      >
        <span
          width="0"
        >
          <span />
          <span>
            This is an exemple how to truncate
          </span>
          <span
            style="position: fixed; visibility: hidden; top: 0px; left: 0px;"
          >
            â€¦
          </span>
        </span>
      </div>
    </div>
    <div
      class="Box-sc-16mv5fl-0 dqYHOz"
      data-testid="Tag"
      display="flex"
      width="fit-content"
    >
      <div
        class="Box-sc-16mv5fl-0 styles__CircleSign-sc-1pvkvyn-0  kPEPzR"
      />
      <div
        class="Box-sc-16mv5fl-0 Text__TextStyled-sc-1eacmib-0 cfMSYc gjXGAy"
        color="blueGray"
        font-size="sm"
      >
        ðŸ§® This is an exemple how to truncate
      </div>
    </div>
  </div>
</div>
`;
